packages/medusa/src/services/price-list.ts

```diff
@@ -0,0 +1,296 @@
+  priceListRepository: typeof PriceListRepository
+  moneyAmountRepository: typeof MoneyAmountRepository
+}
+
+/**
+ * Provides layer to manipulate product tags.
+ * @extends BaseService
+ */
+class PriceListService extends BaseService {
+  private manager_: EntityManager
+  private customerGroupService_: CustomerGroupService
+  private priceListRepo_: typeof PriceListRepository
+  private moneyAmountRepo_: typeof MoneyAmountRepository
+
+  constructor({
+    manager,
+    customerGroupService,
+    priceListRepository,
+    moneyAmountRepository,
+  }: PriceListConstructorProps) {
+    super()
+    this.manager_ = manager
+    this.customerGroupService_ = customerGroupService
+    this.priceListRepo_ = priceListRepository
+    this.moneyAmountRepo_ = moneyAmountRepository
+  }
+
+  withTransaction(transactionManager: EntityManager): PriceListService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new PriceListService({
+      manager: transactionManager,
+      customerGroupService: this.customerGroupService_,
+      priceListRepository: this.priceListRepo_,
+      moneyAmountRepository: this.moneyAmountRepo_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  /**
+   * Retrieves a product tag by id.
+   * @param {string} priceListId - the id of the product tag to retrieve
+   * @param {Object} config - the config to retrieve the tag by
+   * @return {Promise<PriceList>} the collection.
+   */
+  async retrieve(
+    priceListId: string,
+    config: FindConfig<PriceList> = {}
+  ): Promise<PriceList> {
+    const priceListRepo = this.manager_.getCustomRepository(this.priceListRepo_)
+
+    const query = this.buildQuery_({ id: priceListId }, config)
+    const priceList = await priceListRepo.findOne(query)
+
+    if (!priceList) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Price list with id: ${priceListId} was not found`
+      )
+    }
+
+    return priceList
+  }
+
+  /**
+   * Creates a Price List
+   * @param {CreatePriceListInput} priceListObject - the Price List to create
+   * @return {Promise<PriceList>} created Price List
+   */
+  async create(priceListObject: CreatePriceListInput): Promise<PriceList> {
+    return await this.atomicPhase_(async (manager: EntityManager) => {
+      const priceListRepo = manager.getCustomRepository(this.priceListRepo_)
+      const moneyAmountRepo = manager.getCustomRepository(this.moneyAmountRepo_)
+
+      const { prices, customer_groups, ...rest } = priceListObject
+
+      try {
+        const entity = priceListRepo.create(rest)
+
+        if (customer_groups) {
+          const customerGroups: CustomerGroup[] = []
+
+          for (const cg of customer_groups) {
+            const customerGroup = await this.customerGroupService_.retrieve(
+              cg.id
+            )
+            customerGroups.push(customerGroup)
+          }
+
+          entity.customer_groups = customerGroups
```

Comment:
very, very nit: Since `update` has an identical code snippet, it might make sense to refactor into an `upsert` method. If we need to introduce changes in the future, we would only have to do it a single place thereby reducing risk of missing out.